src/java_bytecode/java_entry_point.cpp

```diff
@@ -82,30 +83,74 @@ Function: java_static_lifetime_init
 
 \*******************************************************************/
 
+static bool should_init_symbol(const symbolt& sym)
+{
+  if(sym.type.id()!=ID_code &&
+     sym.is_lvalue &&
+     sym.is_state_var &&
+     sym.is_static_lifetime &&
+     sym.mode==ID_java)
+    return true;
+
+  return has_prefix(id2string(sym.name), "java::java.lang.String.Literal");
```

Comment:
diff shows
```
-  if(has_prefix(id2string(sym.name), "java::java.lang.String.Literal"))
-    return true;
-
-  return false;
+  return has_prefix(id2string(sym.name), "java::java.lang.String.Literal");
```
`has_suffix` is used elsewhere, and I do not see where a suffix `"java::java.lang.String.Literal"` makes sense.